segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/AppendProcessor.java

```diff
@@ -301,6 +303,9 @@ private void handleAppendResult(final Append append, Throwable exception, Timer
                     latestEventNumbers.put(Pair.of(append.getSegment(), append.getWriterId()), append.getEventNumber());
                 } else {
                     if (!conditionalFailed) {
+                        List<Append> appends = waitingAppends.get(append.getWriterId());
+                        pendingBytes -= appends.stream().filter(a -> !a.isConditional())
+                                .mapToInt(a -> a.getData().readableBytes()).sum();
                         waitingAppends.removeAll(append.getWriterId());
```

Comment:
`.removeAll` will return all the items that are removed. You do not need to call `.get` and `.removeAll`; coalesce them into a single call.

Reply:
I have removed this change; because, before calling this method, the getNextAppend is clearing all the data for the give writerid.